fsspec/implementations/asyn_wrapper.py

```diff
@@ -0,0 +1,90 @@
+import asyncio
+import inspect
+import functools
+from fsspec.asyn import AsyncFileSystem
+
+
+def async_wrapper(func, obj=None):
+    """
+    Wraps a synchronous function to make it awaitable.
+
+    Parameters
+    ----------
+    func : callable
+        The synchronous function to wrap.
+    obj : object, optional
+        The instance to bind the function to, if applicable.
+
+    Returns
+    -------
+    coroutine
+        An awaitable version of the function.
+    """
+    @functools.wraps(func)
+    async def wrapper(*args, **kwargs):
+        self = obj or args[0]
+        return await asyncio.to_thread(func, *args, **kwargs)
+    return wrapper
+
+
+class AsyncFileSystemWrapper(AsyncFileSystem):
```

Comment:
This class should appear somewhere in the documentation, even though it's not one we expect users to directly access.